### DISCUSSION TITLE
Source code build instructions?

### DISCUSSION THREAD

**Discussion by nednoodlehead** [czbar/ChessForge#849] [Ideas]
I'm well aware that I can download from source forge, but having some additional instructions for the nerds who want to compile it (and maybe modify it a bit) would be great.

I'm quite unfamiliar with c#, and my attempts at getting it to compile aren't the best lol.

**Comment by czbar** [comment]
Sure, here's how I do it (there are other ways but this definitely works): 
1. If you haven't done this yet: clone the repository to your PC using Git Desktop application.
2. If you don't have it yet: install Visual Studio 2022 Community Edition (which is free). When installing, select items relating to C# Windows development. I can't remember what it asks about exactly but if it asks about .NET versions make sure you include 4.8 (Microsoft made a mess of .NET numbering so 5.0 is something completely different). Also, if there is anything saying "WPF", select that.
3. Open the solution ChessForge.sln in Visual Studio and try building it      

![image](https://github.com/czbar/ChessForge/assets/7270924/d5bb738f-33fb-4f19-8ac4-3ba745c2749d)

If it does not build, then Visual Studio is probably missing some packages. It should tell you which ones and guide you to install them.

Good luck! :)

**Comment by nednoodlehead** [comment]
Yeah, got most the way there. It's just trying to find a dll in a directory that does not exist lol. Not sure if I need to build something to create the dll or if there is something on your side that isn't uploaded to the repo, since the repo doesn't have a \bin\ folder. And 0 results when searching online lol.
![image](https://github.com/czbar/ChessForge/assets/102321690/8116210e-a3a9-4857-8eba-c6ed1cfa4631)

**Reply by czbar** [reply]
ChessPosition.dll is part of the project so it should be built when the solution is compiled.    Does it compile when you try to rebuild it by itself?

![image](https://github.com/czbar/ChessForge/assets/7270924/dadc1ecf-939d-45d5-b58e-13fb25300257)


Also, I notice that it failed in the GenerateOpeningsDictionary project.  That project is a utility that you don't need to build but, yeah, it **should** compile successfully nevertheless.

(Perhaps there is a problem with dependencies and you need to build ChessPosition.dll first before everything else.)
